app/models/place.rb

```diff
@@ -587,24 +585,19 @@ def append_geom(geom, other_attrs = {})
     end
     self.save_geom(new_geom, other_attrs)
   end
-  
-  # Update this place's bbox from a geometry.  Note this skips validations, 
-  # but explicitly recalculates the bbox area
-  def update_bbox_from_geom(geom)
-    self.longitude = geom.centroid.x
-    self.latitude = geom.centroid.y
-    self.swlat = geom.envelope.lower_corner.y
-    self.nelat = geom.envelope.upper_corner.y
-    if geom.spans_dateline?
-      self.longitude = geom.envelope.centroid.x + 180*(geom.envelope.centroid.x > 0 ? -1 : 1)
-      self.swlng = geom.points.map(&:x).select{|x| x > 0 || x < -180}.min
-      self.nelng = geom.points.map(&:x).select{|x| x < 0 || x > 180}.max
-    else
-      self.swlng = geom.envelope.lower_corner.x
-      self.nelng = geom.envelope.upper_corner.x
+
+  def points_from_geom(geom)
+    { latitude: geom.centroid.y, swlat: geom.envelope.lower_corner.y, nelat: geom.envelope.upper_corner.y }.tap do |h|
+      if geom.spans_dateline?
+        h[:longitude] = geom.envelope.centroid.x + 180*(geom.envelope.centroid.x > 0 ? -1 : 1)
+        h[:swlng] = geom.points.map(&:x).select{|x| x > 0 || x < -180}.min
+        h[:nelng] = geom.points.map(&:x).select{|x| x < 0 || x > 180}.max
+      else
+        h[:longitude] = geom.centroid.x,
```

Comment:
Trailing comma is causing `h[:longitude]` to be an array, which I think is the cause of all the spec failures. Behold the marvels of the Ruby interpreter: it has wonderful little gems like `tap` and truly horrific liabilities like this.